RecoEgamma/ElectronIdentification/plugins/ElectronMVAEstimatorRun2Phys14NonTrig.cc

```diff
@@ -34,50 +34,50 @@ ElectronMVAEstimatorRun2Phys14NonTrig::ElectronMVAEstimatorRun2Phys14NonTrig(con
 
 ElectronMVAEstimatorRun2Phys14NonTrig::
 ~ElectronMVAEstimatorRun2Phys14NonTrig(){
-  
-  _tmvaReaders.clear();
 }
 
 float ElectronMVAEstimatorRun2Phys14NonTrig::
-mvaValue( const edm::Ptr<reco::Candidate>& particle){
+mvaValue( const edm::Ptr<reco::Candidate>& particle, const edm::Event& evt) const {
   
-  int iCategory = findCategory( particle );
-  fillMVAVariables( particle );  
-  constrainMVAVariables();
-  float result = _tmvaReaders.at(iCategory)->EvaluateMVA(_MethodName);
+  const int iCategory = findCategory( particle );
+  const std::vector<float> vars = std::move( fillMVAVariables( particle, evt ) );  
+  const float result = _gbrForests.at(iCategory)->GetClassifier(vars.data());
 
-  bool debug = false;
+  constexpr bool debug = false;
   if(debug) {
     std::cout << " *** Inside the class _MethodName " << _MethodName << std::endl;
     std::cout << " bin " << iCategory
-	      << " fbrem " <<  _allMVAVars.fbrem
-         << " kfchi2 " << _allMVAVars.kfchi2
-         << " mykfhits " << _allMVAVars.kfhits
-         << " gsfchi2 " << _allMVAVars.gsfchi2
-         << " deta " <<  _allMVAVars.deta
-         << " dphi " << _allMVAVars.dphi
-         << " detacalo " << _allMVAVars.detacalo
-         << " see " << _allMVAVars.see
-         << " spp " << _allMVAVars.spp
-         << " etawidth " << _allMVAVars.etawidth
-         << " phiwidth " << _allMVAVars.phiwidth
-         << " OneMinusE1x5E5x5 " << _allMVAVars.OneMinusE1x5E5x5
-         << " R9 " << _allMVAVars.R9
-         << " HoE " << _allMVAVars.HoE
-         << " EoP " << _allMVAVars.EoP
-         << " IoEmIoP " << _allMVAVars.IoEmIoP
-         << " eleEoPout " << _allMVAVars.eleEoPout
+	      << " fbrem "    << vars[11]  //_allMVAVars.fbrem
```

Comment:
This for a debug statement. I don't use magic numbers in the rest of the
code.

-L

(Sent from my Nexus 6)
On Aug 7, 2015 19:31, "Slava Krutelyov" <notifications@github.com> wrote:

> In
> RecoEgamma/ElectronIdentification/plugins/ElectronMVAEstimatorRun2Phys14NonTrig.cc
> <https://github.com/cms-sw/cmssw/pull/10618#discussion_r36541165>:
>
> > -         << " mykfhits " << _allMVAVars.kfhits
> > -         << " gsfchi2 " << _allMVAVars.gsfchi2
> > -         << " deta " <<  _allMVAVars.deta
> > -         << " dphi " << _allMVAVars.dphi
> > -         << " detacalo " << _allMVAVars.detacalo
> > -         << " see " << _allMVAVars.see
> > -         << " spp " << _allMVAVars.spp
> > -         << " etawidth " << _allMVAVars.etawidth
> > -         << " phiwidth " << _allMVAVars.phiwidth
> > -         << " OneMinusE1x5E5x5 " << _allMVAVars.OneMinusE1x5E5x5
> > -         << " R9 " << _allMVAVars.R9
> > -         << " HoE " << _allMVAVars.HoE
> > -         << " EoP " << _allMVAVars.EoP
> > -         << " IoEmIoP " << _allMVAVars.IoEmIoP
> > -         << " eleEoPout " << _allMVAVars.eleEoPout
> > +	      << " fbrem "    << vars[11]  //_allMVAVars.fbrem
>
> well, my concern here was mostly manual indexing. Maybe define enums
> instead?
> The current row of vars[magic number] looks pretty bad
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/10618/files#r36541165>.
>